src/Designer/backend/src/Designer/Infrastructure/GitRepository/AltinnOrgGitRepository.cs

```diff
@@ -173,10 +200,36 @@ public async Task UpdateCodeList(string codeListId, List<Option> codeList, Cance
 
         string codeListString = JsonSerializer.Serialize(codeList, s_jsonOptions);
 
-        string codeListFilePath = CodeListFilePath(codeListId);
+        string codeListFilePath = CodeListWithTextResourcesFilePath(codeListId);
         await WriteTextByRelativePathAsync(codeListFilePath, codeListString, false, cancellationToken);
     }
 
+    /// <summary>
+    /// Updates a code list with the provided id.
+    /// </summary>
+    /// <param name="codeListId">The name of the cost list to update.</param>
+    /// <param name="codeList">The code list contents.</param>
+    /// <param name="cancellationToken">A <see cref="CancellationToken"/> that observes if operation is cancelled.</param>
+    public async Task UpdateCodeListNew(string codeListId, CodeList? codeList, CancellationToken cancellationToken = default)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        string codeListFilePath = CodeListFilePath(codeListId);
+
+        if (codeList is null)
+        {
+            if (FileExistsByRelativePath(codeListFilePath))
+            {
+                DeleteFileByRelativePath(codeListFilePath);
+            }
+            return;
```

Comment:
Har dere vurdert å lage en egen funksjon for dette? `DeleteFileByRelativePathIfExists`?